### PR TITLE
[Functions] Remove unused class which didn't get reverted

### PR DESCRIPTION
**Pull request by lhotari**
### Motivation

- this should have been part of #10843 which reverted #8434 changes

### Modifications

remove `pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/PulsarCluster.java` which is unused

### CODE REVIEW

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/PulsarCluster.java [hunk 1]
```diff
@@ -1,84 +0,0 @@
...
-            if (producerSpec.getMaxPendingMessagesAcrossPartitions() != 0) {
-                this.producerBuilder.maxPendingMessagesAcrossPartitions(producerSpec.getMaxPendingMessagesAcrossPartitions());
-            }
-            if (producerSpec.getBatchBuilder() != null) {
-                if (producerSpec.getBatchBuilder().equals("KEY_BASED")) {
-                    this.producerBuilder.batcherBuilder(BatcherBuilder.KEY_BASED);
-                } else {
-                    this.producerBuilder.batcherBuilder(BatcherBuilder.DEFAULT);
-                }
-            }
```
Comment (lhotari) [inline pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/PulsarCluster.java:75]: It's pretty odd that this class happened to contain the logic related to #8523 that was missing from ContextImpl. That problem is fixed by #11706.

**Comment by Anonymitaet** [comment]
@lhotari double check: no need to update docs?

**Comment by lhotari** [comment]
> @lhotari double check: no need to update docs?

@Anonymitaet no need. I added the label.

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/PulsarCluster.java (hunk 1 above)
Comment (lhotari) [reply pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/PulsarCluster.java:75]: I found the explanation: the required code went missing as part of reverting #8434 changes in #10843 . The KEY_BASED logic had been added to the PulsarCluster class after #8434 had been merged.

**Review by eolivelli — APPROVED** [review]
LGTM

**Pull request merged**